test/core/load_balancing/outlier_detection_test.cc

```diff
@@ -453,12 +524,212 @@ TEST_F(OutlierDetectionTest, DoesNotWorkWithPickFirst) {
   // Advance time and run the timer callback to trigger ejection.
   IncrementTimeBy(Duration::Seconds(10));
   LOG(INFO) << "### ejection timer pass complete";
+  
+  // Check that no ejection metrics were recorded (since pick_first doesn't support ejection)
+  EXPECT_THAT(stats_plugin->GetUInt64CounterValue(kEjectionsEnforced, kLabelValues,
+                                                   {"failure_percentage"}),
+              ::testing::Optional(0));
+  
   // Subchannel should not be ejected.
   ExpectQueueEmpty();
   // Subchannel should not see a reconnection request.
   EXPECT_FALSE(subchannel->ConnectionRequested());
 }
 
+TEST_F(OutlierDetectionTest, Metrics) {
```

Comment:
In addition to the unit tests here, please add an end-to-end test in https://github.com/grpc/grpc/blob/master/test/cpp/end2end/xds/xds_outlier_detection_end2end_test.cc that shows that the `grpc.lb.locality` and `grpc.lb.backend_service` labels get properly populated in the xDS case.  You can use the following WRR test as an example:

https://github.com/grpc/grpc/blob/fcf69fba84bf6c2ad3e9ae08a43c29868ac1a7ae/test/cpp/end2end/xds/xds_wrr_end2end_test.cc#L102